<a href="https://colab.research.google.com/github/RaneemIT/Employee-Attrition-DataMining/blob/main/Python%20Notebook/Phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Create the Dataset directory
import os
os.makedirs("Dataset", exist_ok=True)

In [ ]:
import pandas as pd
train_df = pd.read_csv("Dataset/train.csv") # Load and combine train and test sets into one dataframe
test_df = pd.read_csv("Dataset/test.csv")
df=pd.concat([train_df, test_df], ignore_index=True)
# Save the combined data as the required Raw_dataset.csv
df.to_csv("Dataset/Raw_dataset.csv", index=False)
print("Combined dataset shape: ", df.shape)

Combined dataset shape:  (74498, 24)


The combined dataset contains 74,498 instances (rows) and 24 attributes (columns).

The dataset was created by merging the training and testing datasets into one unified dataframe to facilitate data preprocessing and analysis.

In [ ]:
## Project Goal
##The goal of this project is to analyze employee data and apply data mining techniques
##to predict employee attrition using classification and to discover hidden patterns
##among employees using clustering techniques.

In [ ]:
## Dataset Source
##The dataset used in this project was obtained from Kaggle:
##https://www.kaggle.com/datasets/stealthtechnologies/employee-attrition-dataset

In [ ]:
# Read the saved raw dataset
df=pd.read_csv("Dataset/Raw_dataset.csv")
# Display basic dataset information: size and feature count
print("Number of Objects:", df.shape[0])
print("Number of attributes:", df.shape[1])
# Display data types for each attribute
print("\nData types of attributes:\n")
print(df.dtypes)

Number of Objects: 74498
Number of attributes: 24

Data types of attributes:

Employee ID                  int64
Age                          int64
Gender                      object
Years at Company             int64
Job Role                    object
Monthly Income               int64
Work-Life Balance           object
Job Satisfaction            object
Performance Rating          object
Number of Promotions         int64
Overtime                    object
Distance from Home           int64
Education Level             object
Marital Status              object
Number of Dependents         int64
Job Level                   object
Company Size                object
Company Tenure               int64
Remote Work                 object
Leadership Opportunities    object
Innovation Opportunities    object
Company Reputation          object
Employee Recognition        object
Attrition                   object
dtype: object


**Data Types Description**

The dataset contains a mix of numerical and categorical attributes.
 • Numerical attributes (int64) include:
Employee ID, Age, Years at Company, Monthly Income, Number of Promotions, Distance from Home, Number of Dependents, and Company Tenure.
 • Categorical attributes (object) include:
Gender, Job Role, Work-Life Balance, Job Satisfaction, Performance Rating, Overtime, Education Level, Marital Status, Job Level, Company Size, Remote Work, Leadership Opportunities, Innovation Opportunities, Company Reputation, Employee Recognition, and Attrition.

This indicates that the dataset includes both quantitative and qualitative features, which may require encoding during preprocessing.

In [ ]:
# Identify the class label (Attrition) and count instances for each category (Stayed/Left)
print("Class Label: Attrition\n")
print(df["Attrition"].value_counts())

Class Label: Attrition

Attrition
Stayed    39128
Left      35370
Name: count, dtype: int64


**Class Label Description (Attrition)**

The class label in this dataset is Attrition, which indicates whether an employee stayed at the company or left.

**It has two categories:** <br>
 • Stayed: 39,128 employees <br>
 • Left: 35,370 employees

The distribution is relatively balanced, which is suitable for building classification models without major class imbalance issues.

In [ ]:
# Print the first few rows to preview the raw dataset structure
print("Sample of the row dataset:\n")
df.head()

Sample of the row dataset:



Employee ID  Age  Gender  Years at Company    Job Role  Monthly Income  \
0         8410   31    Male                19   Education            5390   
1        64756   59  Female                 4       Media            5534   
2        30257   24  Female                10  Healthcare            8159   
3        65791   36  Female                 7   Education            3989   
4        65026   56    Male                41   Education            4821   

  Work-Life Balance Job Satisfaction Performance Rating  Number of Promotions  \
0         Excellent           Medium            Average                     2   
1              Poor             High                Low                     3   
2              Good             High                Low                     0   
3              Good             High               High                     1   
4              Fair        Very High            Average                     0   

   ... Number of Dependents  Job Level Company Size Company Tenure  \
0  ...                    0        Mid       Medium             89   
1  ...                    3        Mid       Medium             21   
2  ...                    3        Mid       Medium             74   
3  ...                    2        Mid        Small             50   
4  ...                    0     Senior       Medium             68   

   Remote Work Leadership Opportunities Innovation Opportunities  \
0           No                       No                       No   
1           No                       No                       No   
2           No                       No                       No   
3          Yes                       No                       No   
4           No                       No                       No   

   Company Reputation Employee Recognition Attrition  
0           Excellent               Medium    Stayed  
1                Fair                  Low    Stayed  
2                Poor                  Low    Stayed  
3                Good               Medium    Stayed  
4                Fair               Medium    Stayed  

[5 rows x 24 columns]

**Dataset Description Summary**

The combined dataset contains 74,498 records (instances) and 24 attributes (features).

The dataset includes a mixture of numerical and categorical features.
Numerical attributes such as Age, Years at Company, and Monthly Income are stored as integer types (int64).
Categorical attributes such as Gender and other descriptive variables are stored as object type.

The class label in this dataset is Attrition, which represents whether an employee left the company or not.

This dataset will be used for classification to predict employee attrition and for clustering to discover hidden patterns among employees.